client.go

```diff
@@ -13,9 +13,60 @@ type Error struct {
 	Func string
 }
 
+// ProxyNetwork defines the network layer (IPv4/IPv6) a proxy can support
+type ProxyNetwork int
+
+const (
+	// ProxyNetworkAny means the proxy can be used for both IPv4 and IPv6 connections
+	ProxyNetworkAny ProxyNetwork = iota
```

Comment:
do we really want to make proxies `NetworkAny` by default, a better way to handle default/undefined behaviour would be to make the first iota member `ProxyNetworkUndefined` and all users of this enum would return an error if undefined is passed
my 2 cents